### PR TITLE
Comments view: add badge with the number of unresolved comments

### LINKED ISSUE #164623: Comments panel: badge for the number of unresolved comments
Related to #127473, it would be useful to show the number of unresolved comments in a badge.

We currently show the number in the Problems view, but not Comments:
image

This would help users notice they have unresolved comments that needs their attention.

### PR DESCRIPTION
**Pull request by laurentlb**
Fixes #164623

Tested:
  - install the [vscode-pull-request-github](https://github.com/microsoft/vscode-pull-request-github) extension
  - open a pull request that has unresolved comments
  - open the Comments view
  - notice the badge next to the "COMMENTS" title (1 in the screenshot below)

image

### CODE REVIEW

File: src/vs/workbench/contrib/comments/browser/commentsView.ts [hunk 1]
```diff
@@ -99,11 +105,29 @@ export class CommentsPanel extends FilterViewPane implements ICommentsView {
...
 		}));
 
 		this._register(this.commentService.onDidUpdateCommentThreads(e => {
 			this.totalComments += e.added.length;
 			this.totalComments -= e.removed.length;
+
+			let unresolved = 0;
+			for (const resource of this.commentsModel.resourceCommentThreads) {
+				for (const thread of resource.commentThreads) {
+					if (thread.threadState === CommentThreadState.Unresolved) {
```
Comment (alexr00) [inline src/vs/workbench/contrib/comments/browser/commentsView.ts:125]: `this.commentsModel.resourceCommentThreads` has not yet been updated when `onDidUpdateCommentThread` fires, resulting in a low count of unresolved.

**Review by alexr00 — CHANGES_REQUESTED** [review]

File: src/vs/workbench/contrib/comments/browser/commentsView.ts (hunk 1 above)
Comment (laurentlb) [reply src/vs/workbench/contrib/comments/browser/commentsView.ts:125]: Is there another event we could use?

Comment (alexr00) [reply src/vs/workbench/contrib/comments/browser/commentsView.ts]: The problem was that we were listening to the same event twice and so didn't control the ordering. Moving everything into the same listener resolved this.

**Review by alexr00 — APPROVED** [review]
I fixed the undercounting. Thanks for the PR!

**Review by aeschli — APPROVED** [review]

**Pull request merged**